chroma_core/lib/service_config.py

```diff
@@ -337,17 +337,35 @@ def _setup_influxdb(self):
             ]
         )
         self.try_shell(["influx", "-execute", "CREATE DATABASE {}".format(settings.INFLUXDB_IML_STATS_DB)])
-        self.try_shell(
-            [
-                "influx",
-                "-database",
-                settings.INFLUXDB_IML_STATS_DB,
-                "-execute",
-                'CREATE RETENTION POLICY "long_term" ON "{}" DURATION {} REPLICATION 1 SHARD DURATION 5d'.format(
-                    settings.INFLUXDB_IML_STATS_DB, settings.INFLUXDB_IML_STATS_LONG_DURATION,
-                ),
-            ]
-        )
+
+        try:
+            self.try_shell(
+                [
+                    "influx",
+                    "-database",
+                    settings.INFLUXDB_IML_STATS_DB,
+                    "-execute",
+                    'CREATE RETENTION POLICY "long_term" ON "{}" DURATION {} REPLICATION 1 SHARD DURATION 5d'.format(
+                        settings.INFLUXDB_IML_STATS_DB, settings.INFLUXDB_IML_STATS_LONG_DURATION,
+                    ),
+                ]
+            )
+        except CommandError as e:
+            if "retention policy already exists" in e.stderr:
```

Comment:
Maybe just detecting `CommandError` is enough? This could be brittle if the error message ever changes.

Reply:
It could be too broad otherwise - return code is just `1` in case of any error. I can do either way - your call.

Reply:
Just an idea:
```python
p = subprocess.Popen(['influx', '-execute', 'show retention policies on iml_stats', '-format', 'json'], stdout=subprocess.PIPE)
out, _ = p.communicate()
j = json.loads(out)
exists = 'long_term' in  [s[0] for s in j['results'][0]['series'][0]['values']]
if exists:
  # ALTER
else:
  #CREATE
```

Note that ALTER differs from CREATE only in a key word.

Reply:
I agree it casts a wider net, but relying on error strings as an API is too brittle. Just match on the `CommandError`.